modules/hitlnext/src/backend/index.ts

```diff
@@ -18,8 +18,11 @@ export interface StateType {
 
 const state: StateType = { timeouts: {} }
 
-const onServerReady = async (bp: typeof sdk) => {
+const onServerStarted = async (bp: typeof sdk) => {
   await migrate(bp)
+}
+
+const onServerReady = async (bp: typeof sdk) => {
   await upsertAgentRoles(bp)
```

Comment:
Do you think we should also move the `upsertAgentRoles()` and `registerMiddleware()` calls to `onServerStarted()` as well? 

I don't know the inner workings of the `hitlnext` module, but in most other modules, at least `registerMiddleware()` seems to always be called from `onServerStarted()`

Reply:
Not sure about this one. We have to wait that workspaces are loaded properly, thus the onServerReady for this one.